### PR TITLE
feat (api, ui): use the new vcs servers

### PR DESCRIPTION
**Pull request by fsamin**

### CODE REVIEW

File: engine/api/hook/hook.go [hunk 1]
```diff
@@ -247,26 +247,30 @@ func DeleteBranchBuilds(db gorp.SqlExecutor, hooks []sdk.Hook, branch string) er
...
 	t := strings.Split(repoFullName, "/")
 	if len(t) != 2 {
-		return nil, sdk.WrapError(fmt.Errorf("CreateHook> Wrong repo fullname %s.", repoFullName), "")
+		return nil, sdk.WrapError(fmt.Errorf("CreateHook> Wrong repo fullname %s", repoFullName), "")
 	}
 
 	var h sdk.Hook
 
-	h, err = FindHook(tx, application.ID, pipeline.ID, string(rm.Type), rm.URL, t[0], t[1])
+	h, err = FindHook(tx, application.ID, pipeline.ID, rm, rm, t[0], t[1])
```
Comment (yesnault) [inline engine/api/hook/hook.go]: rm , rm ?

File: engine/sql/057_repositoriesmanager_project.sql [hunk 2]
```diff
@@ -0,0 +1,7 @@
+-- +migrate Up
+ALTER TABLE project ADD COLUMN vcs_servers BYTEA;
+ALTER TABLE application ADD COLUMN vcs_server VARCHAR(256);
+
+-- +migrate Down
+DROP TABLE project DROP COLUMN vcs_servers;
```
Comment (yesnault) [inline engine/sql/057_repositoriesmanager_project.sql]: ALTER TABLE and not DROP TABLE

File: engine/api/hook/hook.go (hunk 1 above)
Comment (fsamin) [reply engine/api/hook/hook.go]: Signs point to yes

File: engine/api/bootstrap/bootstrap.go [hunk 3]
```diff
@@ -1,15 +1,22 @@
 package bootstrap
 
 import (
+	"database/sql"
 	"strings"
 
+	"github.com/ovh/cds/engine/api/database/gorpmapping"
```
Comment (sguiheux) [inline engine/api/bootstrap/bootstrap.go]: We would put this import at the end

File: engine/api/bootstrap/bootstrap.go [hunk 4]
```diff
@@ -48,3 +55,89 @@ func InitiliazeDB(defaultValues sdk.DefaultValues, DBFunc func() *gorp.DbMap) er
 
 	return nil
 }
+
+func VCSMigrate(db *gorp.DbMap, cache cache.Store) error {
```
Comment (sguiheux) [inline engine/api/bootstrap/bootstrap.go]: Go doc

File: engine/api/repositories_manager.go [hunk 5]
```diff
@@ -515,11 +515,13 @@ func (api *API) addHookOnRepositoriesManagerHandler() Handler {
 		}
 
 		//Load the repositoriesManager for the project
-		rm := repositoriesmanager.GetVCSServer(proj, rmName)
+		rm := repositoriesmanager.GetProjectVCSServer(proj, rmName)
 		if rm == nil {
 			return sdk.WrapError(sdk.ErrNoReposManager, "attachRepositoriesManager> error loading %s-%s", projectKey, rmName)
 		}
 
+		log.Debug("addHookOnRepositoriesManagerHandler> here %v", rm)
```
Comment (sguiheux) [inline engine/api/repositories_manager.go]: delete

File: engine/vcs/bitbucket/http.go [hunk 6]
```diff
@@ -11,9 +11,10 @@ import (
 	"sort"
 	"time"
 
+	"github.com/ovh/cds/sdk"
```
Comment (sguiheux) [inline engine/vcs/bitbucket/http.go]: at the end

File: engine/vcs/bitbucket/http.go [hunk 7]
```diff
@@ -11,9 +11,10 @@ import (
...
+
 	"github.com/facebookgo/httpcontrol"
 	"github.com/ovh/cds/engine/api/cache"
```
Comment (sguiheux) [inline engine/vcs/bitbucket/http.go]: empty line before

File: ui/src/app/views/application/show/admin/repository/application.repo.component.ts [hunk 8]
```diff
@@ -35,6 +35,7 @@ export class ApplicationRepositoryComponent implements OnInit {
     }
 
     ngOnInit() {
+        console.log(this.application.vcs_server);
```
Comment (sguiheux) [inline ui/src/app/views/application/show/admin/repository/application.repo.component.ts]: remove console.log

**Pull request merged**